File: Java/centaur.java
import java.util.*;

// Gabriel de Jesus Dantas

public class centaur{
    public static void recursive(int a, boolean b){

        //finish
        if(a < 0)
            return;
        
        //call the upper level
        recursive(a-1, false);

        // default values
        String top = "|       #", bottom = "\n";

        //is the first body?
        if (a == 0)
            //+head 
            top = " o\n/|\\_____\n|       #";
        else
            //spaces + default top
            top = (new String("     ")).repeat(a-1) + "! !``" + top;
        
         
        //is the last body?
        if(b)
            //add the legs
            bottom = "\n" + (new String("     ")).repeat(a) + "! !```! !";
        else
            //add the back
            top += "____";

        System.out.print(top + bottom);

    }

    public static void main(String args[]) {
        Scanner scan = new Scanner(System.in);

        int x = scan.nextInt();
        centaur.recursive(x, true);
        
        scan.close();
    }
}

File: Kotlin/centaur.kt
fun centaur(level:Int, finish:Boolean) {
    // finish
    if (level < 0) {
        return
    }
    // call upper level
    centaur(level-1, false)
    // default values
    var top:String = "|       #"
    var bottom:String = ""

    // is head?
    if (level == 0) {
        top = " o\n/|\\_____\n|       #"
    }
    else {
        // spaces + default top
        top = "     ".repeat(level-1) + "! !``" + top
    }
        
    // is last body (legs)?
    if (finish) {
        // add legs
        bottom = "     ".repeat(level) + "! !```! !\n"
    }
    else {
        // add centaur's back
        top += "____"
    }
    print("${top + '\n' + bottom}")
}

fun main() {
    var n:Int = readLine()!!.toInt()

    centaur(n, true)
}